### PR TITLE
feat: add resizeDebounce, maxResizeDebounce

### PR DESCRIPTION
**Pull request by daybrush**
## Issue

#352

### CODE REVIEW

File: src/Watcher.ts [hunk 1]
```diff
@@ -118,13 +116,27 @@ export default class Watcher {
...
+
+			this._debounceTimer = 0;
+			this._resizeTimer = 0;
 			this.resize();
 			this.options.resize();
-			this._timer.resize = null;
-		}, 100);
+		};
+		if (!this._debounceTimer && maxResizeDebounce) {
+			this._debounceTimer = window.setTimeout(onResize, maxResizeDebounce);
```
Comment (mixed) [inline src/Watcher.ts:134]: `_resizeMaxDebounceTimer` rather than `_debounceTimer` seems more good for reading the code. How about?

Comment (mixed) [reply src/Watcher.ts:134]: Logically, `maxResizeDebounce` seems to have meaning only when it is greater than 0 and greater than `resizeDebounce`. How about putting this code in?
```js
if (!this._debounceTimer && maxResizeDebounce > 0 && maxResizeDebounce > resizeDebounce) {
}
```

**Review by mixed — APPROVED** [review]
🥇

File: src/Watcher.ts (hunk 1 above)
Comment (daybrush) [reply src/Watcher.ts:134]: @mixed 

That's good.
I think it would be better to set >= (greater than) rather than >(excess).

**Review by WoodNeck — APPROVED** [review]
LGTM

**Pull request merged**